maths/gcd_of_n_numbers.py

```diff
@@ -87,23 +88,19 @@ def get_greatest_common_divisor(*numbers: int) -> int:
 
     # we just need factors, not numbers itself
     try:
-        data = [get_factors(number) for number in numbers]
+        same_factors, *factors = map(get_factors, numbers)
```

Comment:
https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md#coding-style
> [List comprehensions and generators](https://docs.python.org/3/tutorial/datastructures.html#list-comprehensions) are preferred over the use of lambda, map, filter, reduce

https://docs.python.org/3/whatsnew/3.0.html#views-and-iterators-instead-of-lists